darts/models/forecasting/forecasting_model.py

```diff
@@ -440,20 +440,19 @@ def _get_historical_forecastable_time_index(
             max_future_cov_lag,
         ) = self.extreme_lags
 
-        intersect_ = None
+        if min_target_lag is None:
+            min_target_lag = 0
 
-        # target longest possible time index
-        if (min_target_lag is not None) and (series is not None):
-            intersect_ = generate_index(
-                start=series.start_time()
-                - (min_target_lag - max_target_lag) * series.freq
-                if is_training
-                else series.start_time() - min_target_lag * series.freq,
-                end=series.end_time(),
-                freq=series.freq,
-            )
+        # longest possible time index for target
+        intersect_ = generate_index(
+            start=series.start_time() - (min_target_lag - max_target_lag) * series.freq
+            if is_training
+            else series.start_time() - min_target_lag * series.freq,
+            end=series.end_time(),
+            freq=series.freq,
+        )
 
-        # past covariates longest possible time index
+        # longest possible time index for past covariates
```

Comment:
I'm not sure the past covariates generated indexes cover all use cases:
- What happens with RegressionModel(lags_past_covariates=[-100], output_chunk_length=1)?
 The user supplied `past_covariates` for `fit()`can actually be much shorter than the target `series` (101 steps shorter, I believe).
With the current logic and the shorter `past_covariates`, the intersection would result in a much shorter historic_forecastable_time_index (because of the `end=past_covariates.end_time()`).
The same applies to future_covariates, which can in theory also be used as past_covariates only (lags_future_covariates=[-100]).

I think for future covariates it is correct, so similarly including the max_past_cov lag should probably work